services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java

```diff
@@ -198,6 +199,35 @@ private Map<String, CredentialBuilder> loadCredentialBuilders(KeycloakSession ke
                 .collect(Collectors.toMap(CredentialBuilder::getSupportedFormat, component -> component));
     }
 
+    /**
+     * Validates whether the authenticated client is enabled for OID4VCI features.
+     * <p>
+     * If the client is not enabled, this method logs the status and throws a
+     * {@link CorsErrorResponseException} with an appropriate error message.
+     * </p>
+     *
+     * @throws CorsErrorResponseException if the client is not enabled for OID4VCI.
+     */
+    private void checkClientEnabled() {
+        AuthenticatedClientSessionModel clientSession = getAuthenticatedClientSession();
+        ClientModel client = clientSession.getClient();
+
+        boolean oid4vciEnabled = Boolean.parseBoolean(client.getAttributes().get("oid4vci.enabled"));
+
+        if (oid4vciEnabled) {
+            LOGGER.debugf("Client '%s' is enabled for OID4VCI features.", client.getClientId());
+        } else {
+            LOGGER.debugf("Client '%s' is not enabled for OID4VCI features.", client.getClientId());
+            throw new CorsErrorResponseException(
+                cors,
+                Errors.INVALID_CLIENT,
+                "Client not enabled for OID4VCI",
+                Response.Status.FORBIDDEN
+            );
+        }
```

Comment:
This code is not idiomatically written. Prefer an early return. 

Also, highlight the code you add and auto-format locally with Ctrl+Alt+L. Do not be harsh on line breaks. Ensure "Reformat again to remove custom line breaks" is disabled in your Intellij settings.

```suggestion
        if (!oid4vciEnabled) {
            LOGGER.debugf("Client '%s' is not enabled for OID4VCI features.", client.getClientId());
            throw new CorsErrorResponseException(
                    cors,
                    Errors.INVALID_CLIENT,
                    "Client not enabled for OID4VCI",
                    Response.Status.FORBIDDEN
            );
        }
        
        LOGGER.debugf("Client '%s' is enabled for OID4VCI features.", client.getClientId());
```